Add release factory

jmaupetit opened the pull request:
## Purpose

We need to be able to build multiple e-commerce releases from the same repository similarly to our openedx-docker project.

## Proposal

- [x] add a `releases/` tree
- [x] update docker-compose release in the CI
- [x] update documentation

jmaupetit committed:
♻️(project) improve old ecommerce release compatibility

We are running an old ecommerce fork in production where the
update_assets management command does not exists; errors related to this
command calls are thus ignored.

Similarly, the --no-input management command option does not exist for
old Django releases and has been replaced by its legacy --noinput
equivalent.

jmaupetit committed:
✨(project) add support for multiple releases

Similarly to our openedx-docker project, we've added the possibility to
build images from multiple releases of the ecommerce application from
the same repository/master branch.

sampaccoud commented on .circleci/config.yml:
```diff
@@ -1,4 +1,55 @@
+
+    # Bootstrap
+    - run:
+        name: Bootstrap the CMS & LMS
```
```suggestion
        name: Bootstrap the ecommerce app
```

sampaccoud commented on Dockerfile:
```diff
@@ -95,7 +95,8 @@ COPY --from=front-builder /home/node/ecommerce/static /app/ecommerce/static
 ENV ECOMMERCE_CFG /usr/local/etc/ecommerce/local.yaml
-RUN python manage.py update_assets --skip-collect --settings=ecommerce.settings.production
+RUN python manage.py update_assets --skip-collect --settings=ecommerce.settings.production || \
+      echo "Assets update failed and will be ignored (old release)"
```
can you comment why this is desirable?

sampaccoud commented on Makefile:
```diff
@@ -1,7 +1,10 @@
-EDX_EC_ARCHIVE_URL ?= https://github.com/edx/ecommerce/archive/master.tar.gz
-EDX_EC_RELEASE_REF ?= master
-EDX_EC_DOCKER_TAG  ?= latest
+EDX_EC_RELEASE               ?= master
```
Since it is the name of your image...
```suggestion
EDXEC_RELEASE               ?= master
```
And everywhere else?

jmaupetit replied:
Yup.

jmaupetit committed:
♻️(project) rename EDX_EC_* variables to EDXEC_*

As pointed by Sam, EDXEC is shorter and more consistent with the Docker
image name.
